### PR TITLE
Web layout bootstrap

### PR DESCRIPTION
**Pull request by skaymen**
Significant changes to build an actual web layout for the homepage.

### CODE REVIEW

File: floodviz/static/js/peakflow.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ document.addEventListener("DOMContentLoaded", function(event) {
 	peakflow_bar.style.height = FV.peakmeta['height'];
 	peakflow_bar.style.width = FV.peakmeta['width'];
 
-	var margin = {bottom: 100, right: 120, left: 120, top: 150};
+	var margin = {bottom: 40, right: 40, left: 40, top: 75};
```
Comment (jameeters) [inline floodviz/static/js/peakflow.js]: This isn't a problem with this PR, but these should probably not be set in javascript.

File: floodviz/static/js/peakflow.js [hunk 2]
```diff
@@ -72,12 +72,12 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
-		.attr("x", (width/2))
-		.attr("y", 0 - (margin.top / 2))
-		.attr("text-anchor", "middle")
-		.text(title);
+	// var description = "Lorem ipsum dolor sit amet, consectetur adipiscing elit";
+	// svg.append("text")
+	// 	.attr("x", (width/2))
+	// 	.attr("y", height + 50)
+	// 	.attr("text-anchor", "middle")
+	// 	.text(description);
```
Comment (jameeters) [inline floodviz/static/js/peakflow.js]: This should probably be removed.

File: floodviz/static/css/style.css [hunk 3]
```diff
@@ -1,7 +1,19 @@
 
 html, body, .main-content {
-	height: 100%;
-	margin: 0;
+	/*height: 100%;*/
```
Comment (jameeters) [inline floodviz/static/css/style.css]: This should probably be removed.

File: floodviz/static/css/peakflow.css [hunk 4]
```diff
@@ -1,11 +1,11 @@
 #peakflow_bar {
 	margin: 0 auto;
-	width: 80%;
+	/*width: 80%;*/
 }
 
 #peakflow_bar svg {
 	display: block;
-	margin: auto;
+	/*margin: auto;*/
```
Comment (jameeters) [inline floodviz/static/css/peakflow.css]: This should probably be removed.

File: floodviz/static/css/peakflow.css [hunk 5]
```diff
@@ -1,11 +1,11 @@
 #peakflow_bar {
 	margin: 0 auto;
-	width: 80%;
+	/*width: 80%;*/
```
Comment (jameeters) [inline floodviz/static/css/peakflow.css]: This should probably be removed.

File: floodviz/static/js/peakflow.js (hunk 2 above)
Comment (skaymen) [inline floodviz/static/js/peakflow.js]: Done.

File: floodviz/static/css/style.css (hunk 3 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: Removed.

File: floodviz/static/css/peakflow.css (hunk 5 above)
Comment (skaymen) [inline floodviz/static/css/peakflow.css]: Done.

File: floodviz/static/css/peakflow.css (hunk 4 above)
Comment (skaymen) [inline floodviz/static/css/peakflow.css]: Done.

File: floodviz/static/css/style.css [hunk 6]
```diff
@@ -16,3 +27,66 @@ body {
 	font: 10px sans-serif;
 	stroke-width: 1;
 }
+
+.col-sm-6 {
```
Comment (mbucknell) [inline floodviz/static/css/style.css]: You should not be adding things to bootstrap classes.

File: floodviz/static/css/style.css [hunk 7]
```diff
@@ -1,7 +1,18 @@
 
 html, body, .main-content {
-	height: 100%;
-	margin: 0;
+	margin: 0 auto;
+}
+
+.main-content {
+	max-width: 1200px;
```
Comment (mbucknell) [inline floodviz/static/css/style.css]: I don't think you should be setting this. Let bootstrap handle the width.

File: floodviz/static/css/style.css [hunk 8]
```diff
@@ -16,3 +27,66 @@ body {
...
+.header, .subheader, .footer-icons {
+	text-align: center;
+	border: 1px solid black;
+}
+
+#textbox1 {
+	max-width: 1200px;
+}
+
+#textbox1 {
```
Comment (mbucknell) [inline floodviz/static/css/style.css]: There shouldn't be two #textbox1 styles declarations.

File: floodviz/static/css/style.css [hunk 9]
```diff
@@ -16,3 +27,66 @@ body {
...
+}
+
+.icon {
+	margin-top: 100px;
+	margin-bottom: 100px;
+	height: 25px;
+	width: 25px;
+}
+
+#icons {
```
Comment (mbucknell) [inline floodviz/static/css/style.css]: Not a big fan of having a class name and id name almost the same. How about we call this #icon-div or #icon-container? More descriptive too.

File: floodviz/static/css/style.css (hunk 7 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: Fixed.

File: floodviz/static/css/style.css (hunk 6 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: Fixed.

File: floodviz/static/css/style.css (hunk 8 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: Fixed.

File: floodviz/static/css/style.css (hunk 9 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: Fixed.

File: floodviz/templates/index.html [hunk 10]
```diff
@@ -15,11 +16,97 @@
 
 <body>
 
-	<div class="main-content container-fluid">
-		<div id="hydrograph">
+<div class="header-content container-fluid">
```
Comment (mbucknell) [inline floodviz/templates/index.html]: Indentation needs to be cleaned up in this file.

File: floodviz/templates/index.html [hunk 11]
```diff
@@ -15,11 +16,97 @@
...
+				</div>
+			</div>
+		</div>
+	</div>
+
+	<div class="container-fluid footer">
+		<div class="row">
+			<div class="col-sm-6">
+				<div id="pic3"></div>
+{#				<h1>{{ config.FOOTER | safe }}</h1>#}
```
Comment (mbucknell) [inline floodviz/templates/index.html]: Why is this commented out? Did we conclude from the demo that we should just include some links?

File: floodviz/static/css/style.css [hunk 12]
```diff
@@ -16,3 +24,70 @@ body {
...
 }
+
+#hydrograph_text {
+	margin-left: 50px;
+	text-align: center;
+}
+
+#map_text {
+	text-align: center;
+	/*width: 500px;*/
```
Comment (skaymen) [inline floodviz/static/css/style.css]: Removed.

File: floodviz/static/css/style.css [hunk 13]
```diff
@@ -16,3 +24,70 @@ body {
...
+	margin-left: 50px;
+}
+
+#peakflow_text {
+	/*width: 800px;*/
```
Comment (skaymen) [inline floodviz/static/css/style.css]: Removed.

File: floodviz/static/css/style.css [hunk 14]
```diff
@@ -16,3 +24,70 @@ body {
...
+	border: 1px solid black;
+}
+
+
+#textbox1 {
+	margin-top: 50px;
+}
+
+.icon {
+	/*margin-top: 100px;*/
```
Comment (skaymen) [inline floodviz/static/css/style.css]: Removed.

File: floodviz/static/js/map.js [hunk 15]
```diff
@@ -254,8 +254,10 @@
 				}
 				svg = d3.select(options.div_id)
 					.append('svg')
-					.attr('width', options.width)
-					.attr('height', options.height);
+					// .attr('width', '100%')
```
Comment (skaymen) [inline floodviz/static/js/map.js]: Removed.

File: floodviz/templates/index.html (hunk 10 above)
Comment (skaymen) [inline floodviz/templates/index.html]: Fixed.

File: floodviz/static/js/hydrograph.js [hunk 16]
```diff
@@ -80,8 +80,10 @@
 			// recreate svg
 			svg = d3.select(options.div_id)
 				.append('svg')
-				.attr('width', width + margin.left + margin.right)
-				.attr('height', height + margin.top + margin.bottom)
+				.attr("preserveAspectRatio", "xMinYMin meet")
+				.attr("viewBox", "0 0 " + (width+50) + " " + (height+100))
+				// .attr('width', '100%')
```
Comment (skaymen) [inline floodviz/static/js/hydrograph.js]: Removed.

File: floodviz/static/css/style.css [hunk 17]
```diff
@@ -16,3 +24,70 @@ body {
...
+
+.icon-div:hover {
+	opacity: 1.0;
+}
+
+.icon-div {
+	position: fixed;
+	right: 0;
+	width: 25px;
+	opacity: 0.1;
```
Comment (skaymen) [inline floodviz/static/css/style.css]: Fixed.

File: floodviz/templates/index.html (hunk 11 above)
Comment (skaymen) [inline floodviz/templates/index.html]: Changed to include dummy links.

File: floodviz/static/js/hydrograph.js [hunk 18]
```diff
@@ -80,8 +80,10 @@
 			// recreate svg
 			svg = d3.select(options.div_id)
 				.append('svg')
-				.attr('width', width + margin.left + margin.right)
-				.attr('height', height + margin.top + margin.bottom)
+				.attr("preserveAspectRatio", "xMinYMin meet")
+				.attr("viewBox", "0 0 " + (width+50) + " " + (height+100))
```
Comment (skaymen) [inline floodviz/static/js/hydrograph.js]: Fixed.

File: floodviz/templates/index.html [hunk 19]
```diff
@@ -117,7 +114,7 @@
 		};
 
 		FV.hydrograph_data_path = "{{ url_for('timeseries_data', filename='data/hydrograph_data.json') }}";
-		FV.hydrograph_dimensions = {{ config.CHART_DIMENSIONS | safe }};
+		FV.hydrograph_dimensions = {{ config.HYDROGRAPH_ASPECT_RATIO | safe }};
```
Comment (mbucknell) [inline floodviz/templates/index.html]: You need to make this same change in floodviz/templates/hydrograph.html

File: floodviz/static/css/style.css [hunk 20]
```diff
@@ -26,18 +26,16 @@ body {
 }
 
 #hydrograph_text {
-	margin-left: 50px;
+	/*margin-left: 50px;*/
```
Comment (mbucknell) [inline floodviz/static/css/style.css]: Please remove commented out code.

File: floodviz/static/js/map.js [hunk 21]
```diff
@@ -19,7 +19,8 @@
 		 * specifying config options for the map. Call init() to create the map. Linked
 		 * interaction functions for other figures should be passed to init in and object.
 		 *
-		 * @return {Object} self - Holder for public functions.
+		 * @return {Object} self - Holde
+		 * r for public functions.
```
Comment (mbucknell) [inline floodviz/static/js/map.js]: Not sure why this changed. Please change it back.

File: floodviz/templates/index.html [hunk 22]
```diff
@@ -16,29 +16,27 @@
...
-<div class="header-content container-fluid">
-	<div class ="row">
-		<div class="col-sm-12 header">
-			<img src="static/images/usgsLogo.png"/>
-{#			<h3>{{ config.TITLE | safe }}</h3>#}
+	<div class="header-content container-fluid">
+		<div class ="row">
+			<div class="col-sm-12 header">
+				<img src="static/images/usgsLogo.png"/>
+	{#			<h3>{{ config.TITLE | safe }}</h3>#}
```
Comment (mbucknell) [inline floodviz/templates/index.html]: Why is this commented out?

File: floodviz/templates/index.html [hunk 23]
```diff
@@ -100,11 +98,10 @@
 	<div class="container-fluid footer">
 		<div class="row">
 			<div class="col-sm-6">
-				<div id="pic3"></div>
-{#				<h1>{{ config.FOOTER | safe }}</h1>#}
+				<h5>{{ config.FOOTER | safe }}</h5>
 			</div>
 			<div class="col-sm-6">
-				<div id="pic4"></div>
+				<h5>{{ config.FOOTER | safe }}</h5>
```
Comment (mbucknell) [inline floodviz/templates/index.html]: Not sure why this is in there twice.

File: floodviz/templates/index.html (hunk 22 above)
Comment (skaymen) [inline floodviz/templates/index.html]: Uncommented.

File: floodviz/templates/index.html (hunk 23 above)
Comment (skaymen) [inline floodviz/templates/index.html]: Removed one.

File: floodviz/templates/index.html (hunk 19 above)
Comment (skaymen) [inline floodviz/templates/index.html]: fixed.

File: floodviz/static/css/style.css (hunk 20 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: fixed.

File: floodviz/static/js/map.js (hunk 21 above)
Comment (skaymen) [inline floodviz/static/js/map.js]: Accident, fixed.

**Pull request merged**